lib/HeroBonus.cpp

```diff
@@ -809,8 +802,16 @@ void CBonusSystemNode::updateBonuses(const CSelector &s)
 		child->updateBonuses(s);
 }
 
-void CBonusSystemNode::addNewBonus(Bonus *b)
+void CBonusSystemNode::addNewBonus(const std::shared_ptr<Bonus>& b)

```

Comment:
If we do this, there will be a pointer's copy for sure and unique() will always return false. I am fine with it so I will remove the commented out code in the implementation then.